lib/go/edgecontext/token.go

```diff
@@ -1,13 +1,13 @@
 package edgecontext
 
 import (
+	"github.com/golang-jwt/jwt/v4"
 	"github.com/reddit/baseplate.go/timebp"
-	jwt "github.com/reddit/jwt-go/v3"
 )
 
 // AuthenticationToken defines the json format of the authentication token.
 type AuthenticationToken struct {
-	jwt.StandardClaims
+	jwt.RegisteredClaims
```

Comment:
technically this is a breaking change as the field name from embedding changed. if we want to avoid the breaking change I can switch back to `jwt.StandardClaims` (which is now marked as deprecated in the new version of jwt library).

Reply:
This would break at least one downstream package:
* https://sourcegraph.build.ue1.snooguts.net/github.snooguts.net/reddit/reddit-service-preference-go@d79b044c7f8594447c921609d2d34f67246ad657/-/blob/internal/ectest/loopback.go?L32:24&subtree=true

This is allowed by the baseplate 2022 compatibility promise if we work with any visible breakages to ensure they have a seamless transition.

If it's not a necessary change though, it's probably not worth it.

Reply:
maybe you should upstream ectest to this repo so we can move away from deprecated stuff :) I'll switch back for now.

Reply:
ectest hard-codes some things I don't think we want to put in baseplate.go